src/modal/types.ts

```diff
@@ -1,20 +1,19 @@
 import React from "react";
+import { OverlayClickProps } from "../overlay/types";
 
 export type ModalAnimationDirection = "top" | "bottom" | "left" | "right";
 
-export interface ModalProps extends React.HTMLAttributes<HTMLDivElement> {
+export type ModalProps = React.HTMLAttributes<HTMLDivElement> & {
     show: boolean;
     children: JSX.Element | JSX.Element[];
     /** Animation direction of appearance and dismissal. Values: "top" | "bottom" | "left" | "right" */
     animationFrom?: ModalAnimationDirection | undefined;
-    enableOverlayClick?: boolean | undefined;
     /** The identifier of the element to inject the Modal into */
     rootComponentId?: string | undefined;
     zIndex?: number | undefined;
-    onOverlayClick?: (() => void) | undefined;
     /** Dismiss keyboard to keep modal in fullscreen */
     dismissKeyboardOnShow?: boolean | undefined;
-}
+} & OverlayClickProps;
```

Comment:
unfortunately ModalProps can't inherit `OverlayClickProps` directly because `enableOverlayClick` is true by default in Modal and false by default in Overlay

Reply:
but it's now only inheriting the typing which should be the same..?

Reply:
I was thinking Modal would be like this instead since it defaults to true, i.e. `onOverlayClick` is needed by default. haven't actually tried this though

```
    | {
          enableOverlayClick: false;
          onOverlayClick?: (() => void) | undefined;
      }
    | {
          enableOverlayClick?: true  | undefined;
          onOverlayClick: () => void;
      };
```

Reply:
ah ok i get it. you're right. let me update